rubber_duck.py

```diff
@@ -132,25 +137,92 @@ async def have_conversation(self, thread_id: int, engine: str, prompt: str, init
                         f'\n{ex}\n'
                         '\n'.join(tb.format_exception(ex))
                     )
+
+                    # For server-side errors
+                    if isinstance(ex,
```

Comment:
Instead, introduce a new `except` block before line 131 to catch these specific exceptions: https://docs.python.org/3/tutorial/errors.html#handling-exceptions

Do the same for the other exception groups.